src/Resources.js

```diff
@@ -0,0 +1,132 @@
+(function(context) {
+  'use strict';
+
+  var PersistedResource = {
+    storeType: context.lf.schema.DataStoreType.INDEXED_DB,
+
+    setStoreType: function setStoreType(type) {
+      this.storeType = type;
+
+      return this;
+    },
+
+    setSchemaBuilder: function setSchemaBuilder(schemaBuilder) {
+      this.schemaBuilder = schemaBuilder;
+
+      return this;
+    },
+
+    setTableName: function setTableName(name) {
+      this.tableName = name;
+
+      return this;
+    },
+
+    dbConnection: null,
+
+    getDbConnection: function getDbConnection() {
+      if (this.dbConnection == null) {
+        this.dbConnection = this.schemaBuilder.connect({
+          storeType: this.storeType
+        });
+      }
+
+      return this.dbConnection;
+    },
+
+    getTable: function getTable() {
+      return this.schemaBuilder.getSchema().table(this.tableName);
+    },
+
+    fetch: function fetch(recordUuid) {
+     return this.getDbConnection().then((function(db) {
+       var table = this.getTable();
+        
+       return db.select().from(table).where(table.uuid.eq(recordUuid)).exec();
+      }).bind(this));
+    },
+
+    fetchAll: function fetchAll() {
+      return this.getDbConnection().then((function(db) {
+        return db.select().from(this.getTable()).exec();
+      }).bind(this));
+    }
+  };
+
+  var ResourceCache = Object.create(PersistedResource);
+
+  ResourceCache.createTable = function createTable() {
+    return this.schemaBuilder
+      .createTable(this.tableName)
+      .addColumn('uuid', lf.Type.STRING)
+      .addPrimaryKey(['uuid'])
+      .addColumn('is_dirty', lf.Type.BOOLEAN)
+      .addColumn('created_at', lf.Type.DATE_TIME)
+      .addColumn('updated_at', lf.Type.DATE_TIME);
+  };
+
+  ResourceCache.markClean = function markClean(recordUuids) {
```

Comment:
Is there purpose behind adding these functions (createTable, markClean, fetchAllDirty, etc...) to the ResourceCache instance after its initial creation instead of adding them to the PersistedResource object that was used within the "Object.create" method?